.changeset/tired-carrots-dress.md

```diff
@@ -0,0 +1,18 @@
+---
+'skuba': minor
+---
+
+lint: Add `strictDepBuilds`, `onlyBuiltDependencies`, `trustPolicy`, and `trustPolicyExclude` to `pnpm-workspace.yaml`
+
+These security-focused settings were introduced in pnpm v10 to reduce the risk of installing compromised packages by controlling which packages can execute build scripts and preventing trust downgrades.
+
+`strictDepBuilds` (added in [pnpm v10.3](https://github.com/pnpm/pnpm/releases/tag/v10.3.0)) causes installation to fail if any dependencies have unreviewed build scripts. `onlyBuiltDependencies` specifies which packages are allowed to run "preinstall", "install", and "postinstall" scripts during installation - all other packages will be blocked from running these lifecycle scripts.
+
+`trustPolicy` (added in [pnpm v10.21](https://github.com/pnpm/pnpm/releases/tag/v10.21.0)) can be set to `no-downgrade` to prevent installing package versions where the trust level has decreased compared to previous releases. For example, if a package was previously published by a trusted publisher but now only has provenance or no trust evidence, installation will fail. `trustPolicyExclude` (added in [pnpm v10.22](https://github.com/pnpm/pnpm/releases/tag/v10.22.0)) allows you to bypass this restriction for specific packages or versions.
+
+This change also introduces two new overload fields in `package.json` that allow projects to customize these lists:
```

Comment:
Is there slightly more reason to document this ability than the`minimumReleaseAgeExcludeOverload` do we think?

Reply:
1. I'm curious if these should be nested under the `skuba` key (and/or eventually moved to a `skuba.config.ts`, or if we want them to be generic for interop with e.g. `sku`.
2. How confident are we in our default list? This also has me wanting to look at config dependencies again to avoid an additional layer of abstraction/friction on top of what pnpm supports out of the box (i.e. engineers and agents will try to modify `pnpm-workspace.yaml` by default and it's best to meet them there).